solax/inverters/qvolt_hyb_g3_3p.py

```diff
@@ -59,85 +60,91 @@ def battery_modes(value, *_args, **_kwargs):
         # 44: twoway_div100, almost the same as "41" (battery power),
         # 45: always 1
         # 46: follows PV Output, idles around 44, peaks at 52,
-        'Power Now':                             (47, 'W', to_signed),
+        'Power Now':                             (47, Units.W, to_signed),
         # 48: always 256
         # 49,50: [49] + [50] * 15160 some increasing counter
         # 51: always 5634
         # 52: always 100
         # 53: always 0
         # 54: follows PV Output, idles around 35, peaks at 54,
         # 55-67: always 0
-        'Total Energy':                          (68, 'kWh', total_energy),
-        'Total Energy Resets':                   (69, ''),
+        'Total Energy':                          (68, Total(Units.KWH),
+                                                  total_energy),
+        'Total Energy Resets':                   (69),
         # 70: div10, today's energy including battery usage
         # 71-73: 0
-        'Total Battery Discharge Energy':        (74, 'kWh', discharge_energy),
-        'Total Battery Discharge Energy Resets': (75, ''),
-        'Total Battery Charge Energy':           (76, 'kWh', charge_energy),
-        'Total Battery Charge Energy Resets':    (77, ''),
-        'Today\'s Battery Discharge Energy':     (78, 'kWh', div10),
-        'Today\'s Battery Charge Energy':        (79, 'kWh', div10),
-        'Total PV Energy':                       (80, 'kWh', pv_energy),
-        'Total PV Energy Resets':                (81, ''),
-        'Today\'s Energy':                       (82, 'kWh', div10),
+        'Total Battery Discharge Energy':        (74, Total(Units.KWH),
+                                                  discharge_energy),
+        'Total Battery Discharge Energy Resets': (75),
+        'Total Battery Charge Energy':           (76, Total(Units.KWH),
+                                                  charge_energy),
+        'Total Battery Charge Energy Resets':    (77),
+        'Today\'s Battery Discharge Energy':     (78, Units.KWH, div10),
+        'Today\'s Battery Charge Energy':        (79, Units.KWH, div10),
+        'Total PV Energy':                       (80, Total(Units.KWH),
+                                                  pv_energy),
+        'Total PV Energy Resets':                (81),
+        'Today\'s Energy':                       (82, Units.KWH, div10),
         # 83-85: always 0
-        'Total Feed-in Energy':                  (86, 'kWh', feedin_energy),
-        'Total Feed-in Energy Resets':           (87, ''),
-        'Total Consumption':                     (88, 'kWh', consumption),
-        'Total Consumption Resets':              (89, ''),
-        'Today\'s Feed-in Energy':               (90, 'kWh', div100),
+        'Total Feed-in Energy':                  (86, Total(Units.KWH),
+                                                  feedin_energy),
+        'Total Feed-in Energy Resets':           (87),
```

Comment:
did you miss some 'none' / 'unknown' units here?

Reply:
The mapping is able to handle a variation of fallbacks/defaults 
`'Total Feed-in Energy Resets':           (87),`  is equivalent to 
`'Total Feed-in Energy Resets':           87,`  and
`'Total Feed-in Energy Resets':           (87,Units.UNKNOWN),` or
`'Total Feed-in Energy Resets':           (87,Measurment(Units.UNKNOWN)),`

I did not remove the brackets, so that is somewhat visually consistent